### ISSUE TITLE
ModernTaxonomyPicker can't find term when UI is in language not supported by term store

### ISSUE THREAD

**Issue by Zhephyr54** [pnp/sp-dev-fx-controls-react#1573]
#### Category
[ ] Enhancement

[X] Bug

[ ] Question

#### Version

Please specify what version of the library you are using: [3.13.0]

#### Expected / Desired Behavior / Question
Using the ModernTaxonomyPicker when the SharePoint UI is in a language not supported by the term store (= not in termStoreInfo.languageTags), we should be able to search terms from the text field and find them in the default term store language.

#### Observed Behavior
Using the ModernTaxonomyPicker when the SharePoint UI is in a language not supported by the term store (= not in termStoreInfo.languageTags), while searching terms from the text field, no term can be found at all (the call to the termStore unofficial SP API is returning a 500 error code with the following details: "Specified argument was out of the range of valid values.\r\nParameter name: lcid").

<img width="673" alt="moderntaxonomypicker lang error" src="https://github.com/pnp/sp-dev-fx-controls-react/assets/19250887/d168f145-fd2a-4ea8-bb6e-5b8ca00b45a6">

#### Steps to Reproduce
- Use the ModernTaxonomyPicker with another UI language than english from a SharePoint site where the site collection termstore only supports english language (or with any other language not supported by the termstore)
- Search for any term from the text input
- No result will be return

#### Personal note
Looking at the code, I think it should not be hard to implement a fix.
After the first render, the language used by the component is set like this:
```
const languageTag = props.context.pageContext.cultureInfo.currentUICultureName !== '' ?
          props.context.pageContext.cultureInfo.currentUICultureName :
          currentTermStoreInfo.defaultLanguageTag;
```
As we already have the `currentTermStoreInfo`, we could just check the `currentUICultureName` against the `termStoreInfo.languageTags` array and use `currentTermStoreInfo.defaultLanguageTag` if the current UI language is not supported (= not inside the `termStoreInfo.languageTags` array) by the term store.

Thanks!

**Comment by Zhephyr54** [comment]
I have just submitted a PR to fix this issue:  #1576

**Comment by joelfmrodrigues** [comment]
Merged and will be in the beta release very soon. Please let us know if you find any issues
